libgnucash/quotes/gnc-quotes.cpp

```diff
@@ -0,0 +1,90 @@
+/********************************************************************\
+ * gnc-quotes.hpp -- proxy for Finance::Quote                       *
+ * Copyright (C) 2021 Geert Janssens <geert@kobaltwit.be>           *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+\ *******************************************************************/
+
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <iostream>
+#include <boost/filesystem.hpp>
+#include <boost/process.hpp>
+#include "gnc-quotes.hpp"
+
+extern "C" {
+    #include <glib.h>
+    #include "gnc-path.h"
+}
+
+namespace bp = boost::process;
+
+static GncQuotes quotes_cached;
+
+bool
+GncQuotes::check (void)
+{
+    m_version.clear();
+    m_sources.clear();
+    m_error_msg.clear();
+    m_cmd_result  = 0;
+
+    auto perl_executable = bp::search_path("perl"); //or get it from somewhere else.
+    auto fq_check = std::string(gnc_path_get_bindir()) + "/gnc-fq-check";
+
+    bp::ipstream out_stream;
+    bp::ipstream err_stream;
+    bp::child process (perl_executable, "-w", fq_check, bp::std_out > out_stream, bp::std_err > err_stream);
+
+    std::string stream_line;
+    while (getline (out_stream, stream_line))
```

Comment:
According to the [boost::process tutorial](https://www.boost.org/doc/libs/1_64_0/doc/html/boost_process/tutorial.html#boost_process.tutorial.async_io) you need to protect `getline` with `process.running()`, i.e.
```
    while (process.running() && getline (out_stream, stream_line))
```
because if `process` *isn't* running `getline` will deadlock and freeze up GnuCash. Since this is running during program load debugging it would be impossible.

Reply:
Ok. I copied this from another example on the web, but I think the tutorial should be a better reference.